pkg/asset/ignition/bootstrap/baremetal/template.go

```diff
@@ -97,8 +111,109 @@ func externalURLs(apiVIPs []string) (externalURLv4 string, externalURLv6 string)
 	return
 }
 
+// TLSCertificate contains the private key and certificate contents in []byte format.
+type TLSCertificate struct {
+	privateKey  []byte
+	certificate []byte
+}
+
+func generateTLSCertificate(provisioningIP string) (TLSCertificate, error) {
+	var tlsExpirationDays = 1
+
+	caConfig, err := crypto.MakeSelfSignedCAConfig("metal3-ironic", tlsExpirationDays)
+	if err != nil {
+		return TLSCertificate{}, err
+	}
+
+	ca := crypto.CA{
+		Config:          caConfig,
+		SerialGenerator: &crypto.RandomSerialGenerator{},
+	}
+
+	var host string
+	if provisioningIP == "" {
+		host = "localhost"
+	} else {
+		host = provisioningIP
+	}
+
+	config, err := ca.MakeServerCert(sets.New(host), tlsExpirationDays)
+
+	if err != nil {
+		return TLSCertificate{}, err
+	}
+
+	certBytes, keyBytes, err := config.GetPEMBytes()
+	if err != nil {
+		return TLSCertificate{}, err
+	}
+
+	return TLSCertificate{
+		privateKey:  keyBytes,
+		certificate: certBytes,
+	}, nil
+}
+
+func generateIronicVMediaTLSCertificate(provisioningIP string) error {
+	logrus.Debugf("------Before generating TLS certificate")
+	cert, err := generateTLSCertificate(provisioningIP)
+	if err != nil {
+		return err
+	}
+
+	logrus.Debugf("-------Writing TLS certificate contents for ironic to be used in case TLS for virtual media is enabled")
+	_, filename, _, ok := runtime.Caller(0) // /home/mdevscripts/installer/pkg/asset/manifests/openshift.go
+	if !ok {
+		errStr := "unable to get information about function invocations from the goroutine stack while creating TLS certificate"
+		logrus.Debugf(errStr)
+		return errors.New(errStr)
+	}
+
+	path := filepath.Dir(filepath.Dir(filepath.Dir(filepath.Dir(filename)))) + "/store/data/bootstrap/baremetal/files/etc/ironic"
```

Comment:
This part seems fragile and involves modifying the local checkout. What if you pass both the key and the certificate as strings through TemplateData instead?

Reply:
Its not elegant, for sure. But we need the certificate contents to be in files, so that they can be used in the volumes in the systemd files.

Reply:
You will get files, you just don't the files to be on the installer checkout. See your other patch, you did it nearly right (just don't rely on assets).

Reply:
Like, you don't jump through all these hoops for the newly created `data/data/bootstrap/baremetal/files/etc/containers/systemd/ironic-httpd.container.template`, right? It just works through the installer magic. You can do the same thing with 2 TLS files.